lamagraph-compiler/src/LamaML/Syntax.hs

```diff
@@ -0,0 +1,290 @@
+{- |
+
+LamaML syntax
+-}
+module LamaML.Syntax (
+    -- * Grammar rules
+    -- $rules
+
+    -- * Lexical conventions
+
+    -- ** Blanks
+    -- $lexing_blanks
+
+    -- ** Comments
+    -- $lexing_comments
+
+    -- ** Identifiers
+    -- $lexing_idents
+
+    -- ** Integer literals
+    -- $lexing_int_lits
+
+    -- ** Character literals
+    -- $lexing_char_lits
+
+    -- ** String literals
+    -- $lexing_string_lits
+
+    -- ** Operators
+    -- $lexing_operators
+
+    -- ** Keywords
+    -- $lexing_keywords
+
+    -- * Names
+    -- $names
+
+    -- * Type expressions
+    -- $types
+
+    -- * Constants
+    -- $constants
+
+    -- * Patterns
+    -- $patterns
+
+    -- * Expressions
+    -- $expressions
+
+    -- * Type definitions
+    -- $typedefs
+
+    -- * Declarations
+    -- $decls
+) where
+
+{- $rules
+Grammar rules are written in monospace font.
+Italicized words represent @/nonterminals/@.
+@Terminals@ are written in upright font.
+@( )@ mean grouping, @{ }@ mean zero or more repetitions, @[ ]@ mean optional.
+In case of clash between grammar and terminal meaning, terminal symbol will be typeset as @__bold monospace__@.
+-}
+
+{- $lexing_blanks
+Spaces, tabs, @\\n@ and @\\r@ are ignored and are used to separate tokens.
+-}
+
+{- $lexing_comments
+Comments must be enclosed in @(*@ and @*)@.
```

Comment:
What about single line comment?

Reply:
This may make lexer more complex, but that's all I have against them. 

I think the most natural way to express single-line comment is `--`, just like in Haskell. But that will certainly require tweaking (or maybe even removing) operator rules.

Reply:
Ok. Single-line comments is not a crucial feature.